crates/router/src/services/authentication.rs

```diff
@@ -186,6 +187,7 @@ pub struct SinglePurposeToken {
     pub origin: domain::Origin,
     pub path: Vec<TokenPurpose>,
     pub exp: u64,
+    pub tenant_id: Option<String>,
```

Comment:
What are the problems that this `tenant_id` in the JWT solving?

Is it only because to check whether this token is issued in this `tenant_id` or not?

Is there anything else?

Reply:
Just to check whether the token belong to given tenancy or not.
Token of one tenancy should not be valid in other.

Reply:
Won't this cause conflict the tenant_id in the header?

Reply:
No it will be matched with state, header and state should be same.